src/core/scope.c

```diff
@@ -353,36 +360,78 @@ static void scope_enter_signal(Scope *s, ScopeState state, ScopeResult f) {
         scope_enter_dead(s, SCOPE_FAILURE_RESOURCES);
 }
 
-static int scope_start(Unit *u) {
-        Scope *s = SCOPE(u);
+static int scope_enter_chown_cgroup(Scope *s) {
+        Unit *u = UNIT(s);
+        pid_t pid;
         int r;
 
         assert(s);
```

Comment:
scope_enter_chown_cgroup() expects to be called with s->user set, right?

So we should have an assert for that here.

Alternatively, move the check before the scope_enter_chown_cgroup() invocation into this function, and then make it return 0 if there's no chowning to be done, and > 0 if we are now in the chowning state.

That way, there would be only place to check the setting (i.e. inside this function) instead of two (in the caller via `if` and here via `assert()`).

up to you...

Reply:
I will add an `assert()` and keep the `if` in the caller because I don't want to obscure the fact that we `chown()` the cgroup (i.e. we actually enter the state) only if delegation is turned on and `user` is set.